### DISCUSSION TITLE
PyAnsys Error: Joint Forces

### DISCUSSION THREAD

**Discussion by mredoutey** [ansys/pydpf-core#1105] [Q&A] [answered]
I am trying to extract joint reaction forces from an .rst file and I am getting the following error:

"Source operator "mapdl::rst::JFL" not found"

I can see the joint reaction forces in Ansys Mechanical using a joint probe but when I try to extract them from the rst file I get the error. Is there a way to make sure the joint reactions are written to the rst file?

I am using Ansys 2022R2 and Python 3.9.2. My code is pasted below. 

# Data Sources
path = 'file.rst'
my_data_sources = dpf.DataSources(path)

# Time Scoping
op = dpf.operators.metadata.time_freq_provider()
op.inputs.data_sources.connect(my_data_sources)
my_time_freq_support = op.outputs.time_freq_support.get_data()
my_time_scoping = my_time_freq_support.time_frequencies

# Mesh
op = dpf.operators.mesh.mesh_provider()
op.inputs.data_sources.connect(my_data_sources)
my_mesh = op.outputs.mesh.get_data()

# Scoping
op = dpf.operators.scoping.from_mesh()
op.inputs.mesh.connect(my_mesh)
op.inputs.requested_location.connect(dpf.locations.elemental_nodal)
my_scoping = op.outputs.scoping.get_data()

# Joint Forces
op = dpf.operators.result.joint_force_reaction() 
op.inputs.time_scoping.connect(my_time_scoping)
op.inputs.mesh_scoping.connect(my_scoping)
op.inputs.data_sources.connect(my_data_sources)
op.inputs.mesh.connect(my_mesh)
joint_forces_fc = op.outputs.fields_container.get_data()

Thank you!

**Comment by PProfizi** [comment]
Hi @mredoutey,
Could you please check that DPF detects this result as available with:
```py
print(dpf.Model(my_data_sources))
```
This should list all the metadata found about the `.rst` file, including available results.

**Reply by mredoutey** [reply]
Hi @PProfizi,

The DPF does not detect the joint forces result:

![image](https://github.com/ansys/pydpf-core/assets/138072448/d8a59b75-21fc-4f45-bcfb-6eeaf345fb02)

How can I output that result to the .rst file?

Thank you!

**Reply by PProfizi** [reply]
Hi @ayush-kumar-423,
Could you please help on this topic?
Is it possible to output joint forces to the .rst?

**Comment by ayush-kumar-423** [comment, marked as answer]
@mredoutey, the joint force operator is for Ansys Motion and not for MAPDL results. Sorry for the confusion, we need to improve the documentation here. For MAPDL you can use SMISC results but you need to know the MPC184 Element ID beforehand. I have created posted a solution on Ansys Developer Forum, please refer to this. Please feel free to use Developer Forum for future questions and discussion.

https://discuss.ansys.com/discussion/2442/how-can-i-calculate-joint-forces-on-mpc184-elements-using-pydpf#latest
